File: 0504-Base-7/0504.java
class Solution {
    public String convertToBase7(int num) {
        if (num == 0)
            return "0";

        String ans = "";
        boolean isPositive = num > 0;
        int newNum = Math.abs(num);

        while (newNum > 0) {
            ans = Integer.toString(newNum % 7) + ans;
            newNum /= 7;
        }

        return isPositive ? ans : "-" + ans;
    }

    public String convertToBase7(int num) {
        return Integer.toString(num, 7);
    }
}

File: 0504-Base-7/0504.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0: 
            return '0'

        ans = ""

        is_negative = num < 0
        new_num = abs(num)

        while new_num > 0:
            ans = str(new_num % 7)+ ans
            new_num //= 7
        
        return ans if not is_negative else '-' + ans